### DISCUSSION TITLE
Indexing compounds with non-standard valencies using cartridge

### DISCUSSION THREAD

**Discussion by JamieAtkins** [rdkit/rdkit#5199] [Q&A] [answered]
Hello, apologies if this type of question has been asked before but I am attempting to index, amongst others, the following SMILES string (I have also tried indexing as a MOL)

`[K+].[K+].[Si-2](F)(F)(F)(F)(F)F`

Up to four F atoms works fine but 5 or 6 are rejected as I assume a valency violation has been detected, however I would have thought the -2 on Si would have increased the perceived valency to 6. This is a real substance with CAS no 16871-90-2.

Please could you let me know if there is a way to overcome this? Thank you.

**Comment by greglandrum** [comment, marked as answer]
@Flingoo what specific errors are you seeing?
At least a basic test of this works for me:
```
chembl_30=# select '[K+].[K+].[Si-2](F)(F)(F)(F)(F)F'::mol as m into temporary table foo;
SELECT 1
chembl_30=# insert into foo values ('CCC'::mol);
INSERT 0 1
chembl_30=# insert into foo values ('[Si-2](F)(F)(F)(F)(F)F'::mol);
INSERT 0 1
chembl_30=# select * from foo where m@>'[Si-2]'::qmol;
               m                
--------------------------------
 F[Si-2](F)(F)(F)(F)F.[K+].[K+]
 F[Si-2](F)(F)(F)(F)F
(2 rows)

chembl_30=# create index fooindex on foo using gist(m);
CREATE INDEX
chembl_30=# select * from foo where m@>'[Si-2]'::qmol;
               m                
--------------------------------
 F[Si-2](F)(F)(F)(F)F.[K+].[K+]
 F[Si-2](F)(F)(F)(F)F
(2 rows)
```


Also: which version of the cartridge are you using?

**Reply by JamieAtkins** [reply]
Hello, thanks for your response, I'm using a Docker image provided by one of our developers. Based on your reply I'd imagine we are not using a recent version. I'll speak to the developer tomorrow (it's a public holiday today) and confirm. Apologies for silly question if that's what it turns out to be.

The error I'm seeing is this by the way
`2022-04-18 15:46:21.060 UTC [74] WARNING:  could not create molecule from SMILES '[K+].[K+].[Si-2](F)(F)(F)(F)(F)F'`

**Reply by JamieAtkins** [reply]
Hello, we have rebuilt our Docker image using the Release_2022_03_1 tag, everything seemed to go ok, but we still get the same message above of "could not create molecule from SMILES". Running SELECT rdkit_version(); returns 0.76.0 but I'm not sure how this corresponds to the release versioning.

**Reply by JamieAtkins** [reply]
We have got past this current issue, the problem was that our data insertion script indexed the molecules with this command
update structures set molecule = mol_from_smiles(smiles::cstring)

Looking at the answer we changed this to 
update structures set molecule = structures.smiles::mol
